### PR TITLE
DM-11658: Enable flake8 testing

### PR DESCRIPTION
**Pull request by timj**

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,8 @@
+[flake8]
+max-line-length = 110
+ignore = E133, E226, E228, N802, N803, W503
```
Comment (pschella) [inline setup.cfg]: What are these and why are they ignored?

File: setup.cfg [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+exclude = __init__.py
+
+[tool:pytest]
+addopts = --flake8
+flake8-ignore = E133 E226 E228 N802 N803 W503
```
Comment (pschella) [inline setup.cfg]: Is the repetition required?

Comment (timj) [reply setup.cfg]: Yes. Unfortunately.

File: setup.cfg (hunk 1 above)
Comment (timj) [reply setup.cfg]: They are the standard ignore codes documented in the developer guide.

https://developer.lsst.io/coding/python_style_guide.html#exceptions-to-pep-8

(although two of the ones in the dev guide aren't relevant here: I probably should add them in for consistency).

**Pull request merged**